src/gui/transferlistfilterswidget.cpp

```diff
@@ -549,30 +597,38 @@ void TrackerFiltersList::setDownloadTrackerFavicon(bool value)
     }
 }
 
-void TrackerFiltersList::handleTrackerEntriesUpdated(const QHash<BitTorrent::Torrent *, QHash<QString, BitTorrent::TrackerEntryUpdateInfo>> &updateInfos)
+void TrackerFiltersList::handleTrackerEntriesUpdated(const QHash<BitTorrent::Torrent *, QSet<QString> > &updateInfos)
 {
     for (auto torrentsIt = updateInfos.cbegin(); torrentsIt != updateInfos.cend(); ++torrentsIt)
     {
-        const BitTorrent::TorrentID id = torrentsIt.key()->id();
-        const QHash<QString, BitTorrent::TrackerEntryUpdateInfo> &infos = torrentsIt.value();
+        const BitTorrent::Torrent *torrent = torrentsIt.key();
+        const QSet<QString> &trackerURLs = torrentsIt.value();
+        const BitTorrent::TorrentID id = torrent->id();
 
         auto errorHashesIt = m_errors.find(id);
         auto warningHashesIt = m_warnings.find(id);
 
-        for (auto trackerIt = infos.cbegin(); trackerIt != infos.cend(); ++trackerIt)
+        const QVector<BitTorrent::TrackerEntry> trackers = torrent->trackers();
+        for (const QString &trackerURL : trackerURLs)
         {
-            const QString &trackerURL = trackerIt.key();
-            const BitTorrent::TrackerEntryUpdateInfo &updateInfo = trackerIt.value();
+            const auto iter = std::find_if(trackers.cbegin(), trackers.cend()
```

Comment:
Should this lookup be optimized? by converting `const QVector<BitTorrent::TrackerEntry> trackers` into `QSet` first so you don't need to perform linear lookups (which in turn becomes O(n^2) with a loop).
https://doc.qt.io/qt-5/qset.html#QSet-2

Reply:
> Should this lookup be optimized?

Optimized by inverting things. Now it iterates through `trackers` and lookups in `trackerURLs`.